components/style/properties/shorthand/text.mako.rs

```diff
@@ -40,21 +52,31 @@
         }
 
         Ok(Longhands {
-            text_decoration_color: unwrap_or_initial!(text_decoration_color, color),
             text_decoration_line: unwrap_or_initial!(text_decoration_line, line),
-            text_decoration_style: unwrap_or_initial!(text_decoration_style, style),
+
+            % if product != "servo" or data.testing:
+                text_decoration_style: unwrap_or_initial!(text_decoration_style, style),
+                text_decoration_color: unwrap_or_initial!(text_decoration_color, color),
+            % endif
         })
     }
 
     impl<'a> ToCss for LonghandsToSerialize<'a>  {
         fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
             self.text_decoration_line.to_css(dest)?;
-            dest.write_str(" ")?;
-            self.text_decoration_style.to_css(dest)?;
-            if self.text_decoration_color.parsed != CSSParserColor::CurrentColor {
-                dest.write_str(" ")?;
-                self.text_decoration_color.to_css(dest)?;
-            }
+
+            % if product != "servo" or data.testing:
+                if self.text_decoration_style != &text_decoration_style::get_initial_specified_value() {
```

Comment:
I know that this change was made to make the code generic, but let's hardcode the value here so that we don't have to visit the longhand to know about its initial value (that's what we do for initial values in general IIRC).